mbuild/compound.py

```diff
@@ -1015,13 +1015,15 @@ def add(
                     "outside of the defined simulation box"
                 )
 
-    def remove(self, objs_to_remove):
+    def remove(self, objs_to_remove, reset_labels=False):
```

Comment:
The new behavior with this PR is definitely preferred IMO, so I think we should set `reset_labels=True` by default.

Reply:
Also agree that it will be better to have this flag to set to `True` by default.